microprofile.ls/com.redhat.microprofile.ls/src/main/java/com/redhat/microprofile/services/MicroProfileCompletions.java

```diff
@@ -241,6 +221,58 @@ private static void collectPropertyKeySuggestions(int offset, Node node, Propert
 		}
 	}
 
+	/**
+	 * Collect profiles
+	 * @param offset            the offset where completion was invoked
+	 * @param key               the property key
+	 * @param model             the properties model
+	 * @param markdownSupported boolean determining if markdown is supported
+	 * @param list              the completion list
+	 */
+	private static void collectProfileSuggestions(int offset, PropertyKey key,
+			PropertiesModel model, boolean markdownSupported, CompletionList list) {
+
+		Range range = null;
+		Position currPosition = null;
+		String line = null;
+		TextDocument textDocument = model.getDocument();
+		try {
+			range = textDocument.lineRangeAt(offset);
+			currPosition = textDocument.positionAt(offset);
+			line = textDocument.lineText(currPosition.getLine());
+		} catch (BadLocationException e) {
+			LOGGER.log(Level.SEVERE, "In QuarkusCompletion#collectPropertyKeySuggestions, position error", e);
+			return;
+		}
+
+		// Collect all existing profiles declared in application.properties
+		Set<String> profiles = model.getChildren().stream().filter(n -> n.getNodeType() == NodeType.PROPERTY)
+				.map(n -> {
+					Property property = (Property) n;
+					return property.getProfile();
+				}).filter(Objects::nonNull).filter(not(String::isEmpty)).distinct().collect(Collectors.toSet());
+		// merge existings profiles with default profiles.
+		profiles.addAll(QuarkusModel.getDefaultProfileNames());
+		// Completion on profiles
+		for (String p : profiles) {
+			if (p.equals(key.getProfile())) continue;
+
+			CompletionItem item = new CompletionItem(p);
+			item.setKind(CompletionItemKind.Struct);
+
+			boolean addPeriod = currPosition.getCharacter() < line.length() && line.charAt(currPosition.getCharacter()) != '.';

```

Comment:
This addPeriod is computed for each profiles. It seems it's a global information like, currPosition, line,etc. I suggest you declare this addPeriod after currPosition and you initialize it after currPosition initialisation.